projects/vdk-core/src/vdk/internal/builtin_plugins/version/new_version_check_plugin.py

```diff
@@ -43,48 +43,68 @@ def vdk_configure(self, config_builder: ConfigurationBuilder) -> None:
         )
 
     @hookimpl
-    def vdk_initialize(self, context: CoreContext) -> None:
+    def vdk_exit(self, context: CoreContext) -> None:
         try:
+            package_list = []
             cfg = context.configuration
+
             package_name = cfg.get_value(ConfigKey.PACKAGE_NAME)
             package_index = cfg.get_value(ConfigKey.PACKAGE_INDEX)
-            self.check_version(package_name, package_index)
+            if new_package(package_name, package_index).check():
+                package_list.append(package_name)
 
             if cfg.get_value(ConfigKey.VERSION_CHECK_PLUGINS):
-                log.debug("Will check for newer versions all installed plugins.")
+                log.debug("Will check for newer versions for all installed plugins.")
                 for dist_name, _ in version.list_installed_plugins():
-                    self.check_version(dist_name, package_index, is_plugin=True)
+                    if new_package(dist_name, package_index).check():
+                        package_list.append(dist_name)
+
+            self.check_version(package_list, package_index)
         except Exception as e:
             log.info(
                 f"Could not check for new version release. "
                 f"Error was {e}. We are ignoring the error."
             )
 
     @staticmethod
-    def check_version(package_name, package_index, is_plugin=False):
-        if new_package(package_index, package_name).check():
-            # if package index is not specified that we fetch it from pip repo (pypi.org)
-            extra_index_if_needed = (
-                f"--extra-index-url {package_index}" if package_index else ""
-            )
-            message = (
-                f"New version for a plugin {package_name} is Available"
-                if is_plugin
-                else f"New Version of {package_name} is Available"
-            )
-            # We are using eager strategy so that if there's new version of a plugin that satisfies the requirements
-            # of the main package we want to upgrade to it.
-            pip_command = "pip install --upgrade-strategy eager -U"
-            click.echo(
-                f"""
-                **********************************************************************************************************************
+    def check_version(package_list, package_index):
```

Comment:
If it doesn't need to be public make it protected or private: https://github.com/vmware/versatile-data-kit/wiki/Coding-Standard#clarification-on-access-modifiers

Reply:
done; also included a pydoc for it